### PR TITLE
create ijepa modelcard (ref : PR  #36979 ).

### PR DESCRIPTION
**Pull request by dhruvmalik007**
# What does this PR do?

#36979 : adds the descriptive model card based on the latest standard.

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/ijepa.md [hunk 1]
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
...
-I-JEPA is a self-supervised learning method that predicts the representations of one part of an image based on other parts of the same image. This approach focuses on learning semantic features without relying on pre-defined invariances from hand-crafted data transformations, which can bias specific tasks, or on filling in pixel-level details, which often leads to less meaningful representations.
+[I-JEPA](https://huggingface.co/papers/2301.08243) is a self-supervised learning method that learns semantic image representations by predicting parts of an image from other parts of the image. this works on comparing the abstract representations of the image (rather than pixel level comparisons), which avoids the typical pitfalls of data augmentation bias and pixel-level details that don't capture semantic meaning. You can find the original I-JEPA checkpoints under the [facebook](https://huggingface.co/facebook) organization , includingthe variants based on vairants of vit architecture like [facebook/ijepa_vith14_1k](https://huggingface.co/facebook/ijepa_vith14_1k), [facebook/ijepa_vitg16_22k](https://huggingface.co/facebook/facebook/ijepa_vitg16_22k) etc.
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
[I-JEPA](https://huggingface.co/papers/2301.08243) is a self-supervised learning method that learns semantic image representations by predicting parts of an image from other parts of the image. It compares the abstract representations of the image (rather than pixel level comparisons), which avoids the typical pitfalls of data augmentation bias and pixel-level details that don't capture semantic meaning.

You can find the original I-JEPA checkpoints under the [AI at Meta](https://huggingface.co/facebook/models?search=ijepa) organization.
```

File: docs/source/en/model_doc/ijepa.md [hunk 2]
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
...
 
-The abstract from the paper is the following:
+> [!TIP]
+> This model was contributed by [jmtzt](https://huggingface.co/jmtzt), .
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
> This model was contributed by [jmtzt](https://huggingface.co/jmtzt).
```

File: docs/source/en/model_doc/ijepa.md [hunk 3]
```diff
@@ -66,14 +77,33 @@ def infer(image):
...
 embed_1 = infer(image_1)
 embed_2 = infer(image_2)
 
 similarity = cosine_similarity(embed_1, embed_2)
-print(similarity)
+print(f"Similarity: {similarity.item():.4f}")
 ```
 
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md:91]: Would be helpful to show an quantization example with the large checkpoint https://huggingface.co/facebook/ijepa_vitg16_22k

File: docs/source/en/model_doc/ijepa.md [hunk 4]
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to extract image features with [`Pipeline`] or the [`AutoModel`] class.
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/ijepa_architecture.jpg"
-alt="drawing" width="600"/>
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-<small> I-JEPA architecture. Taken from the <a href="https://huggingface.co/papers/2301.08243">original paper.</a> </small>
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md:44]:
```py
import torch
from transformers import pipeline


feature_extractor = pipeline(
    task="image-feature-extraction",
    model="facebook/ijepa_vith14_1k",
    device=0,
    torch_dtype=torch.bfloat16
)
features = feature_extractor("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg", return_tensors=True)
print(f"Feature shape: {features.shape}")
```

File: docs/source/en/model_doc/ijepa.md [hunk 5]
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
...
+features = feature_extractor("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
+print(f"Feature shape: {len(features[0])}")
+```
 
-Here is how to use this model for image feature extraction:
+</hfoption>
+<hfoption id="AutoModel">
 
-```python
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md:62]:
```py
import requests
import torch
from PIL import Image
from torch.nn.functional import cosine_similarity
from transformers import AutoModel, AutoProcessor

url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
image_1 = Image.open(requests.get(url_1, stream=True).raw)
image_2 = Image.open(requests.get(url_2, stream=True).raw)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", torch_dtype="auto", attn_implementation="sdpa")


def infer(image):
    inputs = processor(image, return_tensors="pt")
    outputs = model(**inputs)
    return outputs.last_hidden_state.mean(dim=1)


embed_1 = infer(image_1)
embed_2 = infer(image_2)

similarity = cosine_similarity(embed_1, embed_2)
print(similarity)
```

File: docs/source/en/model_doc/ijepa.md [hunk 6]
```diff
@@ -66,14 +77,33 @@ def infer(image):
...
 
 similarity = cosine_similarity(embed_1, embed_2)
-print(similarity)
+print(f"Similarity: {similarity.item():.4f}")
 ```
 
+</hfoption>
+</hfoptions>
+
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]: I think we can remove these notes here

File: docs/source/en/model_doc/ijepa.md [hunk 7]
```diff
@@ -66,14 +77,33 @@ def infer(image):
...
+    from transformers import IJepaForImageClassification, AutoProcessor
+
+    model = IJepaForImageClassification.from_pretrained("facebook/ijepa_vith14_1k")
+    processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
+    
+    inputs = processor(images, return_tensors="pt")
+    outputs = model(**inputs)
+    ```
+
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]: Can remove the `## Resources` section as well

File: docs/source/en/model_doc/ijepa.md [hunk 8]
```diff
@@ -14,36 +14,47 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to extract image features with [`Pipeline`] or the [`AutoModel`] class.
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/ijepa_architecture.jpg"
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md:34]: Lets keep the image

**Review by stevhliu — COMMENTED** [review]
Thanks!

File: docs/source/en/model_doc/ijepa.md [hunk 9]
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
...
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with I-JEPA.
-
-<PipelineTag pipeline="image-classification"/>
 
-- [`IJepaForImageClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/image-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/image_classification.ipynb).
-- See also: [Image classification task guide](../tasks/image_classification)
+## Quantization
 
-## IJepaConfig
+The following example demonstrates how to quantize I-JEPA models using bitsandbytes for 4-bit and 8-bit quantization to reduce model size and memory usage while maintaining performance. More quantization methods are discussed [here](https://huggingface.co/docs/transformers/v4.53.2/quantization/overview).
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.

The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

File: docs/source/en/model_doc/ijepa.md [hunk 10]
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
...
 
-[[autodoc]] IJepaConfig
+<hfoptions>
+<hfoption id="Quantization">
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
```

File: docs/source/en/model_doc/ijepa.md [hunk 11]
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
...
-## IJepaForImageClassification
+print(f"Features shape: {features.shape}")
+print(f"Memory usage: {model.get_memory_footprint() / 1e9:.2f} GB")
+```
 
-[[autodoc]] IJepaForImageClassification
-    - forward
\ No newline at end of file
+</hfoption>
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
```

File: docs/source/en/model_doc/ijepa.md [hunk 12]
```diff
@@ -14,85 +14,121 @@ rendered properly in your Markdown viewer.
...
 
-[[autodoc]] IJepaConfig
+<hfoptions>
+<hfoption id="Quantization">
 
-## IJepaModel
-
-[[autodoc]] IJepaModel
-    - forward
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md:97]:
```py
import torch
from transformers import BitsAndBytesConfig, AutoModel, AutoProcessor
from datasets import load_dataset

quantization_config = BitsAndBytesConfig(
    load_in_4bit=True,
    bnb_4bit_quant_type="nf4",
    bnb_4bit_compute_dtype=torch.bfloat16,
    bnb_4bit_use_double_quant=True,
)

url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
image_1 = Image.open(requests.get(url_1, stream=True).raw)
image_2 = Image.open(requests.get(url_2, stream=True).raw)

processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", quantization_config=quantization_config, torch_dtype="auto", attn_implementation="sdpa")


def infer(image):
    inputs = processor(image, return_tensors="pt")
    outputs = model(**inputs)
    return outputs.last_hidden_state.mean(dim=1)


embed_1 = infer(image_1)
embed_2 = infer(image_2)

similarity = cosine_similarity(embed_1, embed_2)
print(similarity)
```

**Review by stevhliu — COMMENTED** [review]
Two more things:

- add the API references back since they shouldn't be removed (`[[autodoc]]`)
- remember to add the image back

File: docs/source/en/model_doc/ijepa.md (hunk 12 above)
Comment (dhruvmalik007) [reply docs/source/en/model_doc/ijepa.md:97]: thanks . sure will do the change In  order to keep consistency.

**Comment by dhruvmalik007** [comment]
> Two more things:
> 
> * add the API references back since they shouldn't be removed (`[[autodoc]]`)
> * remember to add the image back

thanks for reminding the  image.  this was indeed skipped by me .

File: docs/source/en/model_doc/ijepa.md [hunk 13]
```diff
@@ -14,53 +14,109 @@ rendered properly in your Markdown viewer.
...
 url_1 = "http://images.cocodataset.org/val2017/000000039769.jpg"
 url_2 = "http://images.cocodataset.org/val2017/000000219578.jpg"
 image_1 = Image.open(requests.get(url_1, stream=True).raw)
 image_2 = Image.open(requests.get(url_2, stream=True).raw)
 
-model_id = "facebook/ijepa_vith14_1k"
-processor = AutoProcessor.from_pretrained(model_id)
-model = AutoModel.from_pretrained(model_id)
+processor = AutoProcessor.from_pretrained("facebook/ijepa_vith14_1k")
+model = AutoModel.from_pretrained("facebook/ijepa_vith14_1k", quantization_config=quantization_config, torch_dtype="auto", attn_implementation="sdpa")
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
processor = AutoProcessor.from_pretrained("facebook/ijepa_vitg16_22k")
model = AutoModel.from_pretrained("facebook/ijepa_vitg16_22k", quantization_config=quantization_config, torch_dtype="auto", attn_implementation="sdpa")
```

File: docs/source/en/model_doc/ijepa.md [hunk 14]
```diff
@@ -14,53 +14,109 @@ rendered properly in your Markdown viewer.
...
+embed_2 = infer(image_2)  
+
+similarity = cosine_similarity(embed_1, embed_2)  
+print(similarity)
+```
+</hfoption>
+</hfoptions>
+
+
+## Quantization
```
Comment (stevhliu) [inline docs/source/en/model_doc/ijepa.md]:
```suggestion
```

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more things!
